repondeur/zam_repondeur/resources.py

```diff
@@ -0,0 +1,179 @@
+from typing import Iterator, List, Optional
+
+from pyramid.request import Request
+
+from zam_repondeur.models import (
+    Amendement as AmendementModel,
+    Lecture as LectureModel,
+    DBSession,
+)
+
+
+class Resource(dict):
+    """Location-aware resource"""
+
+    __name__: Optional[str] = None
+    __parent__: Optional["Resource"] = None
+
+    @property
+    def breadcrumbs_title(self) -> str:
+        raise NotImplementedError
+
+    breadcrumbs_link: bool = True
+
+    def __init__(self, name: str, parent: "Resource") -> None:
```

Comment:
I wonder if a `dataclass` would work here.

Reply:
Why not. Any benefits you think we could get by using a dataclass here?